osc-server/src/main/java/org/osc/core/broker/service/securitygroup/BindSecurityGroupService.java

```diff
@@ -422,4 +423,31 @@ private ServiceFunctionChain validateAndLoadSfcBind(EntityManager em, BindSecuri
          }
     	return sfc;
     }
+	
+	/**
+     * This methods checks if there exist Security Groups binded with given sfcId and TentantId.
+     * and If there are Security groups, checks if they are binded and not marked for deletion.
+     * throws an exception if all the above conditions are true, declaring service function chain with this virtual system,
+     *  tenant already exist and binded to SecurityGroup.
+     * @param sfcid
+     * @param projectId(TentantId)
+     * @param VirtualSystem
+     * @return
+     */
+    private void checkVirtualSystemRedundancyInServiceFunctionChains(EntityManager em, Long sfcId, VirtualSystem virtualSystem) throws Exception {
```

Comment:
i am not sure about what the method does is correct. It seems like it would only allow an SFC to be used by a single SG within a tenant. We should be able to have an SFC be used by multiple SG...